ext/opentelemetry-ext-otlpexporter/src/opentelemetry/ext/otlpexporter/trace_exporter/__init__.py

```diff
@@ -0,0 +1,305 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""OTLP Span Exporter"""
+
+import logging
+from time import sleep
+from typing import Sequence
+
+from backoff import expo
+from google.rpc.error_details_pb2 import RetryInfo
+from grpc import (
+    ChannelCredentials,
+    RpcError,
+    StatusCode,
+    insecure_channel,
+    secure_channel,
+)
+
+from opentelemetry.proto.collector.trace.v1.trace_service_pb2 import (
+    ExportTraceServiceRequest,
+)
+from opentelemetry.proto.collector.trace.v1.trace_service_pb2_grpc import (
+    TraceServiceStub,
+)
+from opentelemetry.proto.common.v1.common_pb2 import AttributeKeyValue
+from opentelemetry.proto.resource.v1.resource_pb2 import Resource
+from opentelemetry.proto.trace.v1.trace_pb2 import (
+    InstrumentationLibrarySpans,
+    ResourceSpans,
+)
+from opentelemetry.proto.trace.v1.trace_pb2 import Span as CollectorSpan
+from opentelemetry.proto.trace.v1.trace_pb2 import Status
+from opentelemetry.sdk.trace import Span as SDKSpan
+from opentelemetry.sdk.trace.export import SpanExporter, SpanExportResult
+
+logger = logging.getLogger(__name__)
+
+
+# pylint: disable=no-member
+class OTLPSpanExporter(SpanExporter):
+    """OTLP span exporter"""
+
+    def __init__(
+        self,
+        endpoint="localhost:55678",
+        credentials: ChannelCredentials = None,
+        metadata=None,
+    ):
+        super().__init__()
+
+        self._metadata = metadata
+
+        if credentials is None:
+            self._client = TraceServiceStub(insecure_channel(endpoint))
+        else:
+            self._client = TraceServiceStub(
+                secure_channel(endpoint, credentials)
+            )
+
+    # pylint: disable=too-many-locals,too-many-branches,too-many-statements
```

Comment:
It would make reviews easier to break off this method into smaller methods, it would also get rid of this pylint error

Reply:
Fixed